AP_Arming: Add servo voltage check, paramertise the Vcc check

WickedShell opened the pull request:
Needed to prevent arming with extremely bad/questionable servo power supply. The board rail was parameterized on the lower side as it allows the user to specify a higher minimum voltage then the default range, which can be convenient when dealing with backup regulators.

proficnc commented:
I really like this feature! 
@bugobliterator? 
adding to this, there are status flags for checking that the two power rails are good on the cube... having these as part of the prearm would also be good

peterbarker commented on libraries/AP_Arming/AP_Arming.cpp:
```diff
@@ -528,12 +547,24 @@ bool AP_Arming::board_voltage_checks(bool report)
-        if(((hal.analogin->board_voltage() < AP_ARMING_BOARD_VOLTAGE_MIN) || (hal.analogin->board_voltage() > AP_ARMING_BOARD_VOLTAGE_MAX))) {
-            check_failed(ARMING_CHECK_VOLTAGE, report, "Check board voltage");
+        const float bus_voltage =  hal.analogin->board_voltage();
+        if(((bus_voltage < _vbus_min) || (hal.analogin->board_voltage() > AP_ARMING_BOARD_VOLTAGE_MAX))) {
```
Well, you *did* go to the trouble of pulling out a temporary variable.  Suggest reusing it here :-)

peterbarker commented on libraries/AP_Arming/AP_Arming.cpp:
```diff
@@ -528,12 +547,24 @@ bool AP_Arming::board_voltage_checks(bool report)
-            check_failed(ARMING_CHECK_VOLTAGE, report, "Check board voltage");
+        const float bus_voltage =  hal.analogin->board_voltage();
+        if(((bus_voltage < _vbus_min) || (hal.analogin->board_voltage() > AP_ARMING_BOARD_VOLTAGE_MAX))) {
+            check_failed(ARMING_CHECK_VOLTAGE, report, "Board voltage out of range (%1.1f)", bus_voltage);
```
could say, "%1.1f < %1.1f" here.

WickedShell commented:
@peterbarker I struggled a bit with the status text clarity. I've tried to incorporate your idea here, but I presented it a bit differently. (I didn't like the `<` then in the board voltage check, because I would have needed 2 different check_failed's, or a more complicated string)

WickedShell commented:
@Naterater as presented this won't yet help with in flight drop outs.

I'm wondering if the parameters should actually live in AP_BoardConfig and that way AP_Arming and AP_BattMonitor can grab them out of the same place? (AP_BattMonitor currently drives the logging of the `POWR` message, and has the most appropriate failsafes to reuse for this)

OXINARF commented on libraries/AP_Arming/AP_Arming.cpp:
```diff
@@ -528,12 +547,24 @@ bool AP_Arming::board_voltage_checks(bool report)
     }
 #endif
+
+    if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_VOLTAGE)) {
```
We should probably had a new macro for this as the HAL_HAVE_BOARD_VOLTAGE above.

WickedShell commented:
Moved parameters to AP_BoardConfig for future use cases, added a HAL_HAVE_SERVO_VOLTAGE define to make the servo rail check optional.

Most importantly while doing this it was revealed that ChibiOS had disabled the board voltage arming checks due to never having set the define appropriately to allow the check to run. @tridge I'm adding it to the missing chibios list, but once we pull this in it can come right back off.

OXINARF commented on libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:
```diff
@@ -773,6 +773,7 @@ def write_UART_config(f):
             '#define HAL_UART_IO_DRIVER ChibiOS::UARTDriver uart_io(HAL_UART_IOMCU_IDX)\n'
         )
         uart_list.append(config['IOMCU_UART'][0])
+        f.write('#define HAL_HAVE_SERVO_VOLTAGE 1\n') # make the assumption that IO gurantees servo monitoring
```
Nitpicking: guarantees

OXINARF commented:
@davidbuzz Closing doesn't get it merged 😄 

@WickedShell There was a spelling error, but we chose to merge it anyway! Thanks!
